bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linuxsysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linuxsysfs/internal/PollingUsbSerialScanner.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.discovery.usbserial.linuxsysfs.internal;
+
+import static java.lang.Long.parseLong;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDeviceInformation;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDiscovery;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDiscoveryListener;
+import org.eclipse.smarthome.config.discovery.usbserial.linuxsysfs.internal.DeltaUsbSerialScanner.Delta;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link UsbSerialDiscovery} that implements background discovery by doing repetitive scans using a
+ * {@link UsbSerialScanner}, pausing a configurable amount of time between subsequent scans.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "discovery.usbserial.linuxsysfs.pollingscanner")
+public class PollingUsbSerialScanner implements UsbSerialDiscovery {
+
+    private final Logger logger = LoggerFactory.getLogger(PollingUsbSerialScanner.class);
+
+    private static final String THREAD_POOL_NAME = "usb-serial-discovery-linux-sysfs";
+
+    public static final String PAUSE_BETWEEN_SCANS_IN_SECONDS_ATTRIBUTE = "pauseBetweenScansInSeconds";
+    private static final Duration DEFAULT_PAUSE_BETWEEN_SCANS = Duration.ofSeconds(5);
+    private Duration pauseBetweenScans = DEFAULT_PAUSE_BETWEEN_SCANS;
+
+    @NonNullByDefault({})
+    private DeltaUsbSerialScanner deltaUsbSerialScanner;
+
+    private final Set<UsbSerialDiscoveryListener> discoveryListeners = new CopyOnWriteArraySet<>();
+
+    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(THREAD_POOL_NAME);
+    @Nullable
+    private ScheduledFuture<?> backgroundScanningJob;
+
+    @Reference
+    protected void setUsbSerialScanner(UsbSerialScanner usbSerialScanner) {
+        deltaUsbSerialScanner = new DeltaUsbSerialScanner(usbSerialScanner);
+    }
+
+    protected void unsetUsbSerialScanner(UsbSerialScanner usbSerialScanner) {
+        deltaUsbSerialScanner = null;
+    }
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        if (config.containsKey(PAUSE_BETWEEN_SCANS_IN_SECONDS_ATTRIBUTE)) {
```

Comment:
The reason I did not delegate to `modified` here is that `modified` actually does something more than `activate`, viz restarting the background scanning with the new value for the pause between scans.

What you are saying is of course correct, as this restart would simply not be done in `activate`, as the `backgroundScanningJob` is not yet running when `activate` is executed.

I would prefer to leave as is so that it is explicit that background scanning is not restarted on activation.